targets/TARGET_STM/mbed_crc_api.c

```diff
@@ -7,50 +7,103 @@
 
 static CRC_HandleTypeDef current_state;
 static uint32_t final_xor;
-
+static uint32_t crc_mask;
+
+/*  STM32 CRC preipheral features
+  +-------------------------+-----------------------+---------------+---------------+
+  | Feature                 | F1/L1/F2/F4 series    | F0 series     | L0/F3/L4/F7   |
+  +-========================+=======================+===============+===============+
+  | Reversibility option    |          NO           |              YES              |
+  | on I/O data             |                       |                               |
+  +-------------------------+-----------------------+---------------+---------------+
+  | CRC initial Value       | Fixed to 0xFFFFFFFF   | Programmable  | Programmable  |
+  |                         |                       | on 32 bits    | on 8,16,32    |
+  +-------------------------+-----------------------+---------------+---------------+
+  | Handled data size in bit|                  32                   | 8,16,32       |
+  +-------------------------+---------------------------------------+---------------+
+  | Polynomial size in bit  |                  32                   | 7,8,16,32     |
+  +-------------------------+---------------------------------------+---------------+
+  | Polynomial coefficients |         Fixed to 0x4C11DB7            | Programmable  |
+  +-------------------------+---------------------------------------+---------------+
+ */
 bool hal_crc_is_supported(const crc_mbed_config_t *config)
 {
     if (config == NULL) {
         return false;
     }
 
-    if (config->polynomial != POLY_32BIT_ANSI) {
+#if defined(TARGET_STM32F1) || defined(TARGET_STM32F2) || defined(TARGET_STM32F4) || defined(TARGET_STM32L1)
+    if (config->width != HAL_CRC_LENGTH_32B || config->polynomial != POLY_32BIT_ANSI ||
```

Comment:
HAL_CRC_LENGTH_32B is not defined for STM32F1/F2/...,
so use direct value "32"

Reply:
OK, I will push new commit for this.

Reply:
Please verify this commit:
https://github.com/ARMmbed/mbed-os/pull/9347/commits/62a6870ddb516297ef073ed488e0be315fb2aef4

Reply:
I decide set F1/F2/F4/L1 to not supported, more detail please see ba25cd4